src/services/item/plugins/embeddedLink/schemas.ts

```diff
@@ -15,25 +18,27 @@ export const createSchema = S.object()
   .prop('extra', embeddedLinkItemExtraCreate)
   .required(['type', 'extra']);
 
-// equivalent in a static JSON Schema
-// {
-//   type: 'object',
-//   properties: {
-//     type: { const: 'embeddedLink' },
-//     extra: {
-//       type: 'object',
-//       additionalProperties: false,
-//       properties: {
-//         embeddedLink: {
-//           type: 'object',
-//           additionalProperties: false,
-//           properties: {
-//             url: { type: 'string', format: 'url' }
-//           },
-//           required: ['url']
-//         }
-//       },
-//       required: ['embeddedLink']
-//     }
-//   }
-// }
+// schema defining the allowed properties for the link item
+export const updateExtraSchema = S.object()
+  .additionalProperties(false)
+  .prop(
+    ItemType.LINK,
+    // only allow to set url in extra on create
+    S.object().additionalProperties(false).prop('url', S.string().format('url')).required(['url']),
+  )
+  .required([ItemType.LINK]);
+
+export const getLinkMetadata = {
+  querystring: S.object().additionalProperties(false).prop('link', S.string().format('url')),
```

Comment:
Could it be interesting to extract the object schema in a const and reuse it in the three schema ?